packages/stark-ui/src/modules/minimap/components/minimap.component.ts

```diff
@@ -0,0 +1,173 @@
+ */
+const _cloneDeep: Function = require("lodash/cloneDeep");
+
+export type StarkMinimapComponentMode = "compact";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-minimap";
+
+/**
+ * Component to display a minimap which permits to toggle the visibility of passed elements.
+ * The minimap shows the label of the elements to display with a checkbox to enable/disable the visibility:
+ *
+ * @param string -mode Desired layout or flavour:
+ *
+ *    - compact: displayed in a compact form. There is only a button to trigger the dropdown.
+ *    - default: basic implementation with everything
+ *
+ * @param StarkMinimapItemProperties[] - items Array of StarkMinimapItemProperties objects which define the items to display in the minimap.
+ *
+ * @param string[] - visibleItems Array of names of the items that are visible.
+ *
+ * @param showHideItem EventEmitter to be called when call showHideItem callback.
+ *
+ */
+@Component({
+	selector: "stark-minimap",
+	templateUrl: "./minimap.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkMinimapComponent implements OnInit, OnChanges, OnDestroy {
+	@Input()
+	public mode?: StarkMinimapComponentMode;
+	@Input()
+	public parentItems: StarkMinimapItemProperties[];
+	@Input()
+	public parentVisibleItems: string[];
+
+	@Output()
+	public showHideItem: EventEmitter<StarkMinimapItemProperties> = new EventEmitter<StarkMinimapItemProperties>();
+
+	public items: StarkMinimapItemProperties[];
+	public visibleItems: string[];
+
+	public windowClickHandler: EventListener;
+	public isDisplayedMenu: boolean;
+
+	public constructor(@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService, private ngZoneService: NgZone) {}
+
+	public ngOnInit(): void {
+		this.visibleItems = _cloneDeep(this.parentVisibleItems) || [];
+		this.items = _cloneDeep(this.parentItems) || [];
+
+		this.isDisplayedMenu = false;
+		this.attachWindowClickHandler();
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnChanges(changesObj: SimpleChanges): void {
+		if (changesObj["parentItems"] && !_isEqual(this.parentItems, changesObj["parentItems"].previousValue)) {
+			this.logger.debug(componentName + ": items changed... Recreating items config...", changesObj["parentItems"]);
+			this.items = _cloneDeep(this.parentItems);
+		}
+
+		if (changesObj["parentVisibleItems"] && !_isEqual(this.parentVisibleItems, changesObj["parentVisibleItems"].previousValue)) {
+			this.logger.debug(
+				componentName + ": visibleItems changed... Recreating visibleItems config...",
+				changesObj["parentVisibleItems"]
+			);
+			this.visibleItems = _cloneDeep(this.parentVisibleItems);
+		}
+	}
+
+	/**
+	 * Component lifecycle hook
+	 */
+	public ngOnDestroy(): void {
+		this.logger.debug(componentName + ": removing clickHandler");
+		this.removeWindowClickHandler();
+	}
+
+	/**
+	 * Retrieve the label of an item
+	 * @param StarkMinimapItemProperties - the item whose label we want to retrieve
```

Comment:
I think this JSDoc should be like this: `@param item - The item whose label we want to retrieve`... because the type is already implicit in the code ;)